Move Stylo to its own repo

delan opened the pull request:
This patch moves Stylo and several of its dependencies to [servo/stylo](https://github.com/servo/stylo).

---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `___` with appropriate data: -->
- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors

<!-- Either: -->
- [ ] There are tests for these changes OR
- [x] These changes do not require tests because there are no functional changes

delan committed:
Remove packages that were moved to external repo

delan committed:
Add workspace dependencies pointing to 2023-06-14 branch

delan committed:
Fix servo-tidy.toml errors

mrobinson commented:
> Test results for linux-wpt-layout-2020 from try job ([#8014145026](https://github.com/servo/servo/actions/runs/8014145026)):
> Flaky unexpected result (12)
> Stable unexpected results that are known to be intermittent (21)
> Stable unexpected results (92)

Stylo will just need a few changes included from the current Servo version for these tests to pass.

delan committed:
Update commit to include #31346

delan committed:
Update commit to include servo/stylo#2

delan committed:
Move css-properties.json lookup to target/doc/stylo

delan committed:
Remove dependency on vendored mako in favour of pypi dependency

This also removes etc/ci/generate_workflow.py, which has been unused
since at least 9e71bd6a7010d6e5723831696ae0ebe26b47682f.

delan committed:
Add temporary code to debug Windows test failures

delan committed:
Fix failures on Windows due to custom target dir

delan committed:
Update commit to include servo/stylo#3

delan committed:
Fix license in tests/unit/style/build.rs

delan committed:
Document how to build with local Stylo in Cargo.toml
